### PR TITLE
PP-448 Added Integration tests for UserNotificationService running against stubs created

### PR DESCRIPTION
**Pull request by rahul-vinayak**
Notify email integration tests which run against wiremock stub for notify.

### CODE REVIEW

**Comment by ijmad** [comment]
SSL issues, I can help with, but SSL needs to work end to end before we should merge this.

**Comment by rahul-vinayak** [comment]
@ijmad now that we have changed the strategy of the notify integration tests in this PR from running against the Stubs to running against wiremocks, the above SSL issue is not relevant anymore. The E2E tests that will be running against the Notify Stubs though will still have the SSL issue to look into but thats a different PR anyways.

File: src/test/java/uk/gov/pay/connector/service/UserNotificationServiceIntTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Test
+    public void notifyPaymentFailedEmailRequest() throws Exception {
+        notifyEmailMock.responseWithEmailRequestResponse(400, BAD_REQUEST_RESPONSE, -1);
+        Optional<String> idOptional = userNotificationService.notifyPaymentSuccessEmail("mail@stubbed.com");
+        assertFalse(idOptional.isPresent());
+    }
+
+    @Test(expected = GovNotifyClientException.class)
```
Comment (imranbohoran) [inline src/test/java/uk/gov/pay/connector/service/UserNotificationServiceIntTest.java]: Minor: We seem to be using the old style of expected here. We have been using `ExpectedException` rule in other places. The expected rule is handy if you want to check for a specific message in the exception. If you think its useful here, lets put it. If not I'm happy with this too.

Comment (rahul-vinayak) [inline src/test/java/uk/gov/pay/connector/service/UserNotificationServiceIntTest.java]: I wont say its an old style, depends what you want your test to do, could make use of that but its thrown by the notify client library and is not with any message even, so just this basic level exception check (with fewer lines) serves the purpose I believe.

**Comment by imranbohoran** [comment]
👍

**Pull request merged**